### ISSUE TITLE
Devcontainer generator with --dev flag unitialized constant.

### ISSUE THREAD

**Issue by thejonroberts** [rails/rails#52704]
### Steps to reproduce
```sh
rails new example_app --dev
cd example_app
bin/rails g devcontainer --dev 
```

### Expected behavior
Generates  .devcontainer directory files with no error (and create mount points for local rails, as in `rails new example_app --dev --devcontainer`)

### Actual behavior
```sh
[rails_path]/railties/lib/rails/generators/rails/devcontainer/devcontainer_generator.rb:157:in `local_rails_mount': 
uninitialized constant Rails::Generators::RAILS_DEV_PATH (NameError)
```

### System configuration
**Rails version**:
Rails 8.0.0.alpha
**Ruby version**:
3.3.4

NOTE: I think that this may have something to do with #51880 - `Rails::Generators::RAILS_DEV_PATH` no longer available in new location?

I've made a quick PR, but it's my first dive into the codebase. There may be a better way to do it, but it seems to work.

**Comment by jeromedalbert** [comment]
The issue also happens on Rails main:

```shell
$ rails new myapp --main
$ cd myapp
$ bin/rails g devcontainer --dev

      create  .devcontainer
/Users/jerome/.asdf/installs/ruby/3.3.2/lib/ruby/gems/3.3.0/bundler/gems/rails-c6e3336cfdec/railties/lib/rails/generators/rails/devcontainer/devcontainer_generator.rb:145:in `local_rails_mount': uninitialized constant Rails::Generators::RAILS_DEV_PATH (NameError)

            source: Rails::Generators::RAILS_DEV_PATH,
                                     ^^^^^^^^^^^^^^^^

```

**Comment by jeromedalbert** [comment]
> NOTE: I think that this may have something to do with https://github.com/rails/rails/pull/51880

This is correct, that PR introduced the devcontainer generator and command, with commit f31b6f02eafc18eec6a58d5055aeb4aa1c271da2 introducing the generator. I checked out that commit and it looks like `bin/rails g devcontainer --dev` has never worked; only `bundle exec rails new myapp --devcontainer --dev` has.

**Comment by thejonroberts** [comment]
PR is in a pretty good spot now. However...

It seems like the app generator or `bin/rails devcontainer` command is preferred over the generator because it will pick up app configuration automatically (very cool! see the PR mentioned above or [7.2 changelog](https://github.com/rails/rails/blob/7-2-stable/railties/CHANGELOG.md#rails-720-august-09-2024)).

As such, the generator was not directly tested for this usage - although it is very well tested from the app generation side. I've started copying & modifying those tests into the test file I added for my fix PR.  I'll keep it separate from this issue. We could create a new (or convert this issue to) "Test the devcontainer generator".

My opinion is that it should either be directly tested or hidden from the user via `Rails::Generators#hidden_namespaces`, curious what others think.
